File: backend/app/routers/analytics.py
from fastapi import APIRouter, Depends
from ..database import get_db
from ..dependencies import verify_admin
from datetime import datetime, timedelta

router = APIRouter()

@router.get("/admin-dashboard")
async def get_dashboard(admin_user=Depends(verify_admin), db=Depends(get_db)):
    cursor = db.cursor()
    
    cursor.execute("SELECT COUNT(*) as c, SUM(price) as t FROM orders WHERE status='completed'")
    row = cursor.fetchone()
    total_revenue = row['t'] or 0.0
    total_orders = row['c'] or 0
    
    cursor.execute("SELECT COUNT(*) as c FROM users")
    total_users = cursor.fetchone()['c'] or 0
    
    today_start = datetime.now().replace(hour=0, minute=0, second=0).strftime("%Y-%m-%d %H:%M:%S")
    cursor.execute("SELECT SUM(price) as t FROM orders WHERE status='completed' AND created_at >= ?", (today_start,))
    today_revenue = cursor.fetchone()['t'] or 0.0
    
    cursor.execute("SELECT COUNT(*) as c FROM orders WHERE status='pending'")
    pending_orders = cursor.fetchone()['c'] or 0
    
    return {
        "success": True,
        "dashboard": {
            "total_revenue": total_revenue,
            "today_revenue": today_revenue,
            "total_orders": total_orders,
            "pending_orders": pending_orders,
            "total_users": total_users,
            "total_games": 0 # Not heavily tracked in MVP dashboard
        }
    }

File: backend-node/src/routes/analytics.js
const express = require('express');
const router = express.Router();
const db = require('../db/helpers');
const { verifyAdmin } = require('../middleware/auth');

// GET /api/admin-dashboard
router.get('/admin-dashboard', verifyAdmin, async (req, res) => {
  try {
    const completedOrders = await db.get("SELECT COUNT(*) as c, SUM(price) as t FROM orders WHERE status = 'completed'");
    const totalRevenue = completedOrders ? (completedOrders.t || 0.0) : 0.0;
    const totalOrders = completedOrders ? (completedOrders.c || 0) : 0;

    const userCount = await db.get("SELECT COUNT(*) as c FROM users");
    const totalUsers = userCount ? (userCount.c || 0) : 0;

    const todayStr = new Date().toISOString().split('T')[0] + ' 00:00:00';
    // Handle both ISO time and simple date strings by checking >= start of today
    const todayRevenueRow = await db.get(
      "SELECT SUM(price) as t FROM orders WHERE status = 'completed' AND datetime(created_at) >= datetime(?)",
      [todayStr]
    );
    const todayRevenue = todayRevenueRow ? (todayRevenueRow.t || 0.0) : 0.0;

    const pendingRow = await db.get("SELECT COUNT(*) as c FROM orders WHERE status = 'pending'");
    const pendingOrders = pendingRow ? (pendingRow.c || 0) : 0;

    const gamesRow = await db.get("SELECT COUNT(*) as c FROM games");
    const totalGames = gamesRow ? (gamesRow.c || 0) : 0;

    return res.status(200).json({
      success: true,
      dashboard: {
        total_revenue: totalRevenue,
        today_revenue: todayRevenue,
        total_orders: totalOrders,
        pending_orders: pendingOrders,
        total_users: totalUsers,
        total_games: totalGames
      }
    });
  } catch (error) {
    console.error('Get dashboard stats error:', error);
    return res.status(500).json({ success: false, error: 'Internal Server Error' });
  }
});

module.exports = router;
